src/main/scala/nl/knaw/dans/api/sword2/servlets/ServiceInitializer.scala

```diff
@@ -0,0 +1,103 @@
+    DepositHandler.startDepositProcessingStream(settings)
+    log.info("Initialization completed.")
+  }
+
+  def initLogging(homeDir: File): Unit = {
+    try {
+      LoggerFactory.getILoggerFactory match {
+        case lc: LoggerContext => {
+          val logConfigFile = new File(homeDir, "cfg/logback.xml")
+          val configurator = new JoranConfigurator
+          configurator.setContext(lc)
+          lc.reset()
+          configurator.doConfigure(logConfigFile)
+          val log = LoggerFactory.getLogger(getClass)
+          log.info(s"Home directory = $homeDir")
+          log.info(s"Logback configuration file = $logConfigFile")
+        }
+        case _ => println("Logback not found. Could not configure logging")
+      }
+    } catch {
+      // Printing this, because servlet container log message is sometimes very terse
+      case t: Throwable => println(s"Error during logging configuration $t"); throw t
+    }
+  }
+
+  def readSettings(homeDir: File): Settings = {
+    val log = LoggerFactory.getLogger(getClass)
+    val config = {
+      val ps = new PropertiesConfiguration()
+      ps.setDelimiterParsingDisabled(true)
+      ps.load(new File(homeDir, "cfg/application.properties"))
+      ps
+    }
+    val depositRootDir = new File(config.getString("deposits.rootdir"))
+    if (!depositRootDir.canRead) throw new ServletException("Cannot read deposits dir")
+    val depositPermissions = config.getString("deposits.permissions")
+    val tempDir = new File(config.getString("tempdir"))
+    if (!tempDir.canRead) throw new ServletException("Cannot read tempdir")
+    val baseUrl = config.getString("base-url")
+    val collectionIri = config.getString("collection.iri")
+    val auth = config.getString("auth.mode") match {
+      case "ldap" => LdapAuthSettings(new URI(config.getString("auth.ldap.url")),
+        config.getString("auth.ldap.users.parent-entry"),
+        config.getString("auth.ldap.sword-enabled-attribute-name"),
+        config.getString("auth.ldap.sword-enabled-attribute-value"))
+      case "single" => SingleUserAuthSettings(config.getString("auth.single.user"), config.getString("auth.single.password"))
+      case _ => throw new RuntimeException(s"Invalid authentication settings: ${config.getString("auth.mode")}")
+    }
+    val urlPattern = Pattern.compile(config.getString("url-pattern"))
+    // TODO: make use of bag-store optional through configuration
+    val bagStoreBaseUri = config.getString("bag-store.base-url") // TODO: make File, check existence
+    val bagStoreBaseDir = config.getString("bag-store.base-dir") // TODO: make File, check existence
+    val supportMailAddress = config.getString("support.mailaddress")
+    Settings(
+      depositRootDir, depositPermissions, tempDir, baseUrl, collectionIri, auth, urlPattern,
+      bagStoreBaseUri, bagStoreBaseDir, supportMailAddress)
+  }
+
+  def contextDestroyed(sce: ServletContextEvent) = Unit
```

Comment:
If you're not gonna use this method, don't return an arbitrary type 😉 . Just say: `... = {}` or something like that.